drv/sidecar-front-io/src/transceivers.rs

```diff
@@ -695,38 +977,59 @@ impl Transceivers {
     /// Waits for all of the I2C busy bits to go low
     ///
     /// Returns a set of masks indicating which channels (among the ones active
-    /// in the input mask) have recorded FPGA errors.
+    /// in the input mask) have recorded FPGA errors. The meaning of the
+    /// returned `ModuleResult`:
+    /// success: the desired I2C transaction completed successfully
+    /// failure: there was an I2C error
+    /// error: an `FpgaError` occurred
     pub fn wait_and_check_i2c(
         &mut self,
-        mask: FpgaPortMasks,
-    ) -> Result<FpgaPortMasks, FpgaError> {
-        let mut out = FpgaPortMasks::default();
+        mask: LogicalPortMask,
+    ) -> ModuleResult {
+        let mut physical_failure: FpgaPortMasks = FpgaPortMasks::default();
+        let mut physical_error: FpgaPortMasks = FpgaPortMasks::default();
```

Comment:
I don't think you need to specify the types here, since they're obvious on the RHS.
```suggestion
        let mut physical_failure = FpgaPortMasks::default();
        let mut physical_error = FpgaPortMasks::default();
```
(unless you want symmetry with the line below, for _A E S T H E T I C S_)